packages/astro/src/core/errors/errors-data.ts

```diff
@@ -530,6 +530,27 @@ See https://docs.astro.build/en/guides/server-side-rendering/ for more informati
 			)} are supported for optimization.`,
 		hint: "If you do not need optimization, using an `img` tag directly instead of the `Image` component might be what you're looking for.",
 	},
+	/**
+	 * @docs
+	 * @see
+	 * - [`getStaticPaths()`](https://docs.astro.build/en/reference/api-reference/#getstaticpaths)
+	 * - [`params`](https://docs.astro.build/en/reference/api-reference/#params)
+	 * @description
+	 * The endpoint is prerendered with an `undefined` param but the generated file and folder names will collide.
+	 *
+	 * An additional extension can be added to the endpoint file name to generate the file with a different name. For example, renaming `pages/api/[slug].ts` to `pages/api/[slug].json.ts`.
```

Comment:
```suggestion
		 * The endpoint is prerendered with an `undefined` param so the generated path will collide with another route.
	 *
	 * If you cannot prevent passing `undefined`, then an additional extension can be added to the endpoint file name to generate the file with a different name. For example, renaming `pages/api/[slug].ts` to `pages/api/[slug].json.ts`.
```
So much better, folks! Would you say this slight tweak is still accurate? @bluwy @wulinsheng123 (Just also looking below at the message which suggests that not passing `undefined` is a way out of this error.)

Reply:
Sounds good to me too 👍